### PR TITLE
Nightmare Insert: Trijent Dam - Crashed UPP Survivors

### PR DESCRIPTION
**Pull request by Steelpoint**
# About the pull request

**Starring**
_The UPP 173rd Airborne Reconnaissance Company, 2nd Platoon_
Screenshot 2023-07-09 16 10 32

This PR adds a new nightmare insert to Trijent Dam that sees a small group of UPP survivors, post a crash landing in their Mi-220 Assault Gunship "Fulcrum", attempting to survive the xenomorph outbreak. Whilst successful in holding their ground despite their initial casualties, the group are now running low on ammunition and their equipment is starting to falter. Rescue is coming, but from whom? The Americans? Is the enemy of my enemy really our friend?

---------------

Technically speaking the PR will see the deployment of the following survivor roles
* 4 Riflemen
* 1 Non-Combat Synthetic
* 2 Sappers
* 2 Medics
* 1 Specialist
* 1 Squad Leader

Screenshot 2023-07-10 01 54 11
Squad Patch

The UPP are in dire straits and this is represented in game by them being spawned with a mish mash of armour and items. Only the Specialist and SL are guaranteed to spawn with full armour, everyone else is likely getting clothing or webbing. If a soldier spawns with webbing they will be given a flashlight in their suit storage slot. 

Weapon's wise it is a even 33% chance for each role to either get a UPP Pulse Rifle, Pulse Carbine or a UPP Shotgun, with each only getting two spare mags/3 shell handfuls. The Specialist is guaranteed to get the UPP Pulse Rifle/Flamer. The mini-gun in the DS is a prop and non-functional. 

The area is littered with mostly props or junk, the 'HEAP' ammo mags are all empty. There is a single ammo box of soft-point ammo set to around 120 rounds. 

Every UPP member speaks Russian by default, with Chinese and German as their secondary languages. Only the SL and Synth can speak English. 

The Synthetic has lesser equipment compared to its ERT counterpart and uses survivor synth skills. It has similar combat restrictions to its USCM counterpart. 

The lore has been run past the lore channel and has been given a thumbs up. 

-----------
**Other Additions**

- External webbing can be clipped to UPP uniforms, which is used to facilitate some of the cosmetic options for the survivors. 
- New UPP logo has been added for documents and faxes. Official UPP documents now can look more official (see images for example).
- UPP synthetics will now speak Russian as their first language instead of English

-----------

The PR (as far as I can tell) works and runs properly. I mostly have a few other tasks of moderate difficulty to finish.
To-Do
- [x] Add the UPP synth
- [x] Figure out how to get the UPP to spawn with random UPP names
- [x] Sprite the dropships side and rear wings
- [x] Expand the in-game lore and descriptions
- [x] Get spawn chances set to correct odds (20% ?)
- [ ] BALANCE

# Explain why it's good for the game

I believe that this would add further variety to the game in two respects.

Firstly, the UPP are a somewhat under-used aspect that generally only appear during events or as hostile ERT. Technically in-universe the USCM and UPP are in a cold war and are technically not at war, I believe that their introduction as a uncommon survivor spawn may provide some interesting opportunities for roleplay to occur, either from actual encounters between UPP and USCM personnel, or from the USCM finding the last stand of the UPP. 

Secondly I think more variety in survivor roles is a boon for the game, and the opportunity to play as a UPP soldier in a non-ERT setting may present more opportunities for UPP roleplay to occur. 

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

In game map as it stands

Screenshot 2023-07-09 16 10 32
Overview of the UPP as they stand

Screenshot 2023-07-10 01 54 11
Squad Patch

Screenshot 2023-07-11 12 09 54
Your Orders + UPP logo

</details>

# Changelog
:cl:
add: A team of UPP survivors may spawn on Trijent Dam. 
mapadd: A new nightmare insert for Trijent Dam has been added, featuring a crashed UPP dropship.
/:cl:

### CODE REVIEW

**Comment by Steelpoint** [comment]
If the addition of UPP as survivors is not something that can be accepted, I'll gladly reorient this PR to just add the crashed UPP ship as a purely cosmetic nightmare insert, with no UPP survivors spawning.

File: code/modules/clothing/under/marine_uniform.dm
Comment (BeagleGaming1) [inline code/modules/clothing/under/marine_uniform.dm:537]: Why have this be 1-4 if 1-2 are one option and 3-4 are another? why not just 1-2?

File: code/modules/clothing/under/marine_uniform.dm
Comment (BeagleGaming1) [inline code/modules/clothing/under/marine_uniform.dm:555]: same for this one

File: code/modules/clothing/under/marine_uniform.dm
Comment (BeagleGaming1) [inline code/modules/clothing/under/marine_uniform.dm:573]: and this one

File: maps/Nightmare/maps/BigRed/nightmare.json
Comment (BeagleGaming1) [inline maps/Nightmare/maps/BigRed/nightmare.json:12]: seems you disabled the chance for other inserts to spawn here

Comment (Steelpoint) [reply maps/Nightmare/maps/BigRed/nightmare.json:12]: This is for my testing purposes. I'm also not sure what odds this should spawn in for.

File: code/modules/clothing/under/marine_uniform.dm
Comment (Steelpoint) [reply code/modules/clothing/under/marine_uniform.dm:537]: good points

**Comment by MusselMain** [comment]
maybe ensure that all UPP survivors are equipped with some form of lightsource, as unless things have changed since i checked a year or two ago, UPP pulse rifles and carbines cannot mount any form of light, which usually isn't an issue, but could be an issue in a setting where half of the survivors have no armour and therefore no lightsource.

**Comment by Steelpoint** [comment]
@QuickLode I set the UPP synth to be a non-combatant only as a balance precaution, if it is deemed acceptable for it to be a combat synthetic I can attempt to make the change.

@MusselMain I've already done that. Any UPP that does not spawn with a set of combat armour will instead spawn with external webbing and a flashlight in their suit storage slot.

**Comment by Steelpoint** [comment]
The main outstanding thing I am trying to figure out but am struggling with is getting the UPP to spawn with random UPP names, they can keep their regular appearance. I'd also like the synth when spawned to only overwrite a Rifleman spawn and not take one of the more valuable slots like SL or Spec. 

Everything else is just lore and balancing the spawn.

**Comment by Steelpoint** [comment]
I'll leave it to the maintainers to state the availability of ammo and weapons, the numbers are akin to the more military themed survivors. 

I could do a similar Sorokyne version of this PR in the future on that note @QuickLode though I don't see a reason why the UPP can't appear on Big Red, especially as there is a in-universe reason the UPP may be interested in the area, which I've even detailed in a lore briefing document.

**Comment by Steelpoint** [comment]
I've updated the UPP synth to have no firearm restrictions. It has no armour equipped but it spawns with a PK-9 just to emphasize to the player they are able to use firearms. Its loadout is limited and heavily infers they should be a support role and not a active combat role. 

Screenshot 2023-07-10 14 47 04

**Comment by Steelpoint** [comment]
Trying to get the names sorted. Synths spawns with random UPP name, still trying to find out how to get the same to work for the humans.  (mapper doing coding work)

The survivors will retain their user made appearance but they should get UPP style random names. So no one questions why Bill Carson or Malcolm is working for the UPP.

**Review by morrowwolf — CHANGES_REQUESTED** [review]
The reason the CLF surv gets firearm restriction off is because the hold is almost impossible and CLF are pretty much obliged to be anti-USCM which puts them between both factions rather than a boon to either/or.

Essentially: UPP synth needs firearm restrictions.

File: code/modules/clothing/under/marine_uniform.dm
Comment (realforest2001) [inline code/modules/clothing/under/marine_uniform.dm:543]: Creating a subtype just to randomise if the suit is rolled in places or not is a little excessive/bloat.

File: code/modules/clothing/under/marine_uniform.dm
Comment (realforest2001) [inline code/modules/clothing/under/marine_uniform.dm:561]: Above

File: code/modules/clothing/under/marine_uniform.dm
Comment (realforest2001) [inline code/modules/clothing/under/marine_uniform.dm:579]: Above

File: code/modules/gear_presets/survivors.dm
Comment (realforest2001) [inline code/modules/gear_presets/survivors.dm:1515]: These procs should be on `/datum/equipment_preset` rather than `/datum/equipment_preset/survivor/upp`

File: code/modules/gear_presets/synths.dm
Comment (realforest2001) [inline code/modules/gear_presets/synths.dm:477]: Use /datum/equipment_preset/synth/survivor/upp

File: code/modules/gear_presets/synths.dm
Comment (realforest2001) [inline code/modules/gear_presets/synths.dm:479]: While I'm not a balance reviewer, I can tell you with confidence a survivor combat synth has a next to zero chance of being included, even if it is a rare spawn.

**Review by realforest2001 — CHANGES_REQUESTED** [review]

**Comment by Steelpoint** [comment]
Combat restrictions are back in place. We can just state that not all UPP Synthetics have combat restrictions rescinded.

File: code/modules/clothing/under/marine_uniform.dm
Comment (Steelpoint) [reply code/modules/clothing/under/marine_uniform.dm:543]: Is there an alternative way for me to get the uniforms to start with the sleeves or jacket rolled?

Comment (morrowwolf) [reply code/modules/clothing/under/marine_uniform.dm:543]: In load_gear() of the gear set create the object, do your random check, change it on the uniform, then equip it.

Comment (Steelpoint) [reply code/modules/clothing/under/marine_uniform.dm:543]: Morrow, you might one day make a coder out of me.

**Comment by Steelpoint** [comment]
UPP now get random names on spawn, exception of Synths. I've set the weight odds for the nightmare insert, I'll need feedback if the numbers are fine.

Outside of balancing and further tidy up, the PR is good to go I feel.

**Comment by Foxtrot13224** [comment]
You might want to make it so the synth spawns in with Russian set as their main language, as all synths spawn in with English set to their primary language, I am willing to bet someone will roll UPP synth and then not realise their comrades can't speak English, making them basically useless to the other survs.

**Comment by Steelpoint** [comment]
> You might want to make it so the synth spawns in with Russian set as their main language, as all synths spawn in with English set to as primary language, I am willing to bet someone will roll UPP synth and then not realise their comrades can't speak English, making them basically useless to the other survs.

Good point. 

I've set it that the Synth will now have Russian as their first language of choice.

I've also retroactivly set the standard UPP ERT Synth to also speak Russian as their first language.

**Comment by Steelpoint** [comment]
As per team request I will be moving the insert from Big Red to Trijent Dam.

**Comment by Steelpoint** [comment]
I've reworked the UPP survivors skillset to be more in-line with military survivors.

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by Steelpoint** [comment]
All requested code changes have been made.

**Review by harryob — CHANGES_REQUESTED** [review]

File: maps/map_files/DesertDam/standalone/crashlanding-upp-bar.dmm
Comment (Nanu308) [inline maps/map_files/DesertDam/standalone/crashlanding-upp-bar.dmm:1]: bilde
Noop turfs and Areas.
bilde
APC missing, throw it in a corner somewhere.

Other than that, mapping lgtm. 👍

**Review by Nanu308 — CHANGES_REQUESTED** [review]
Mapping

**Comment by Steelpoint** [comment]
Requested changes made.

File: maps/map_files/DesertDam/standalone/crashlanding-upp-bar.dmm
Comment (Nanu308) [inline maps/map_files/DesertDam/standalone/crashlanding-upp-bar.dmm:1]: lgtm

**Review by harryob — APPROVED** [review]

**Pull request merged**
